### ISSUE TITLE
manager not transmitted to pulsar?

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/pulsar#105]
I tried setting up 2 named managers to select from, but I'm struggling here.
Any chance this is not implemented for non-message queue pulsar?

```
File "/Users/marius/pulsar/pulsar/pulsar/web/routes.py", line 39, in _app_args
    app_args['manager'] = managers[manager_name]
KeyError: '_default_'
```

With the corresponding job_conf.xml looking like this:
```
<job_conf>
    <plugins>
         <plugin id="local" type="runner" load="galaxy.jobs.runners.local:LocalJobRunner" workers="4"/>
         <plugin id="pulsar" type="runner" load="galaxy.jobs.runners.pulsar:PulsarRESTJobRunner"/>
    </plugins>
    <handlers>
        <handler id="main"/>
    </handlers>
    <destinations default="local">
        <destination id="local" runner="local">
         <env file="/Users/marius/src/galaxy/.venv/bin/activate" />
        </destination>
        <destination id="pulsar_quv" runner="pulsar">
            <param id="url">http://127.0.0.1:8913/</param>
            <param id="manager">quv</param>
            <param id="jobs_directory">/work/vandenbeek/staging</param>
            <param id="file_action_config">/Users/marius/src/galaxy/config/pulsar_action_quv.yml</param>
        </destination>
        <destination id="pulsar_qice" runner="pulsar">
            <param id="url">http://127.0.0.1:8913/</param>
            <param id="manager">qice</param>
            <param id="jobs_directory">/lustrework/vandenbeek/staging</param>
            <param id="file_action_config">/Users/marius/src/galaxy/config/pulsar_action_qice.yml</param>
        </destination>
        </destinations>
    <tools>
        <tool id="bowtie2" destination="pulsar_qice" />
        <tool id="printenv" destination="pulsar_quv" />
	</tools>
</job_conf>
```
 and this in app.yml:
```
---
managers:
  quv:
    type: queued_cli
    job_plugin: Torque
    shell_plugin: SecureShell
    shell_hostname: mesu0.dsi.upmc.fr
    shell_username: vandenbeek
    job_Resource_List: "select=1:ncpus=16:mem=32gb"
    job_User_List: "vandenbeek"
    job_destination: "quv"
    staging_directory: /work/vandenbeek/staging

  qice:
    type: queued_cli
    job_plugin: Torque
    shell_plugin: SecureShell
    shell_hostname: mesu0.dsi.upmc.fr
    shell_username: vandenbeek
    job_Resource_List: "select=1:ncpus=16:mem=32gb"
    job_User_List: "vandenbeek"
    job_destination: "qice"
    staging_directory: /lustrework/vandenbeek/staging

job_metrics_config_file: /Users/marius/pulsar/job_metrics_conf.xml
```

**Comment by jmchilton** [comment]
I think manager is only used by MQ destinations. I think you want to change the URL - maybe - ``http://127.0.0.1:8913/managers/qice``?

This surprised me too the other day when I was walking around the code - I'm not sure if manager ever worked for URLs or what the documentation says though.

**Comment by mvdbeek** [comment]
Hmm, that just gives me (on the galaxy side):
```
galaxy.jobs.runners.pulsar ERROR 2016-04-15 16:37:29,415 failure running job 235
Traceback (most recent call last):
  File "/Users/marius/src/galaxy/lib/galaxy/jobs/runners/pulsar.py", line 228, in queue_job
    job_id = pulsar_submit_job(client, client_job_description, remote_job_config)
  File "/Users/marius/src/galaxy/lib/pulsar/client/staging/up.py", line 22, in submit_job
    file_stager = FileStager(client, client_job_description, job_config)
  File "/Users/marius/src/galaxy/lib/pulsar/client/staging/up.py", line 96, in __init__
    self.__upload_input_files()
  File "/Users/marius/src/galaxy/lib/pulsar/client/staging/up.py", line 165, in __upload_input_files
    self.__upload_input_file(input_file)
  File "/Users/marius/src/galaxy/lib/pulsar/client/staging/up.py", line 171, in __upload_input_file
    self.transfer_tracker.handle_transfer(input_file, path_type.INPUT)
  File "/Users/marius/src/galaxy/lib/pulsar/client/staging/up.py", line 365, in handle_transfer
    response = self.client.put_file(path, type, name=name, contents=contents, action_type=action.action_type)
  File "/Users/marius/src/galaxy/lib/pulsar/client/client.py", line 181, in put_file
    return self._upload_file(args, contents, input_path)
  File "/Users/marius/src/galaxy/lib/pulsar/client/decorators.py", line 13, in replacement
    response = func(*args, **kwargs)
  File "/Users/marius/src/galaxy/lib/pulsar/client/client.py", line 242, in _upload_file
    return self._raw_execute("upload_file", args, contents, input_path)
  File "/Users/marius/src/galaxy/lib/pulsar/client/client.py", line 215, in _raw_execute
    return self.job_manager_interface.execute(command, args, data, input_path, output_path)
  File "/Users/marius/src/galaxy/lib/pulsar/client/interface.py", line 96, in execute
    response = self.transport.execute(url, method=method, data=data, input_path=input_path, output_path=output_path)
  File "/Users/marius/src/galaxy/lib/pulsar/client/transport/standard.py", line 34, in execute
    response = self._url_open(request, data)
  File "/Users/marius/src/galaxy/lib/pulsar/client/transport/standard.py", line 20, in _url_open
    return urlopen(request, data)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1227, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
URLError: <urlopen error [Errno 32] Broken pipe>
```

**Comment by mvdbeek** [comment]
Ah damnit, you gave me the right hint, and I didn't read/copy_paste carefully ...
instead of 
```<param id="url">http://127.0.0.1:8913/managers/quv</param>```
I put 
```<param id="url">http://127.0.0.1:8913/manager/quv</param>``` .
Well, it works now. Should I add this as an example to job_conf.xml in galaxy, or should I transform the url in HttpPulsarInterface ?

**Comment by mvdbeek** [comment]
Closed via #108 and #104
